### ISSUE TITLE
How to convert sigma level vertical velocity to pressure level?

### ISSUE THREAD

**Issue by weatherforecasterwhai** [google-research/neuralgcm#69]
Thanks to @shoyer in #68 , I've got sigma level vertical velocity, however, the output is on sigma levels.
So I try to use **model.model_coords.horizontal.to_nodal(vertical_velocity)** as @shoyer  said in https://github.com/google-research/neuralgcm/issues/8.

However it works with this error:
**_size of label '1' for operand 1 (65) does not match previous terms (64)._**
I've searched and can't fix it by myself. The full traceback is like this:

![1715263857330](https://github.com/google-research/neuralgcm/assets/162259387/11e618a6-36a5-49ca-8b26-40b198005ce4)

**Comment by shoyer** [comment]
I believe to_nodal and to_modal only operate on 2D arrays. If you want to transform a higher dimensional field, wrap the transformations in jax.vmap.

**Comment by weatherforecasterwhai** [comment]
Thanks to mention the dimension.
Then advanced.state.temperature_variation.shape is (32,128,65), and the vertical velocity shape is (32,128,64).
So, maybe the to_nodal can operate on 3D array (32,128,65), but can't operate on (32,128,64), and the error lies between 65 and 64.

**Comment by weatherforecasterwhai** [comment]
@shoyer I don't understand the codes whether it outputs sigma level or pressure level. Please have a look at compute_vertical_velocity:
_**def compute_vertical_velocity(
    state: State, coords: coordinate_systems.CoordinateSystem
) -> jax.Array:
  """Calculate vertical velocity at the center of each layer."""
  sigma_dot_boundaries = compute_diagnostic_state(state, coords).sigma_dot_full
  assert sigma_dot_boundaries.ndim == 3
  # This matches the default boundary conditions for vertical velocity
  # from sigma_coordinates.centered_vertical_advection
  sigma_dot_padded = jnp.pad(sigma_dot_boundaries, [(1, 1), (0, 0), (0, 0)])
  return 0.5 * (sigma_dot_padded[1:] + sigma_dot_padded[:-1])**_

**Comment by shoyer** [comment]
Vertical velocity (and everything inside the dynamical core) is output on sigma levels. If you want it on pressure levels, you can use the vertical interpolation routines from `dinosaur.vertical_interpolation`.

**Comment by weatherforecasterwhai** [comment]
Vertical velocity is really important to analyse weather, please make a easy api to output it in the same way as p_e or other variables. Thank you!

**Comment by shoyer** [comment]
Another option is to calculate vertical velcoity as a diagnosed variable from an output dataset on pressure levels with the WeatherBench2 code: https://weatherbench2.readthedocs.io/en/latest/_autosummary/weatherbench2.derived_variables.VerticalVelocity.html#weatherbench2.derived_variables.VerticalVelocity
